instance/models/server.py

```diff
@@ -287,8 +289,8 @@ def public_ip(self):
 
         try:
             public_addr = openstack.get_server_public_address(self.os_server)
-        except novaclient.exceptions.NotFound:
-            return None  # This server has been deleted.
+        except Exception:  # pylint: disable=broad-except
```

Comment:
We can definitely adjust that. I chose to go with a broad `except Exception` here because novaclient fails to catch all the exceptions that come out of Requests - we'd need to be able to catch a very broad range of possible exceptions. We could do that explicitly, though; I just felt that the relatively "thin" layer of remaining code underneath this `except` that's actually owned by us made the tradeoff worthwhile.